---
title: Tolerance of hybrid hosts against infections, TAC Meeting December 2022
author:
  - name: Fay Webster
    email: fay.webster@hu-berlin.de
    corresponding_author: TRUE # Automatic footnote using corresponding_author
    affiliation:
      - ABC
      - DEF
    #biography: 
    #biography_pic: "path"
    #biography_pic_wid/PATH/TO/image.pngth: "1in"
  - name: Lubomír Bednář
    email: bednarlu@hu-berlin.de
    affiliation:
      - ABC
      - DEF
    footnote:  # Footnote using reference
      - 2
  - name: Emanuel Heitlinger
    email: emanuel.heitlinger@hu-berlin.de
    affiliation: 
      - ABC
      - DEF
    footnote:
        - 1
address:
  - code: ABC
    department: Department of Molecular Parasitology, Institute for Biology
    organization: Humboldt University Berlin (HU)
    street: Philippstraße 13
    #postcode: 
    city: Berlin
    state: 10115 # Mix and match elements to get proper address
    country: Germany
  - code: DEF
    department: Research Group Ecology and Evolution of Molecular Parasite-Host Interactions
    organization: Leibniz-Institut for Zoo and Wildlife Research (IZW)
    street: Alfred-Kowalke-Straße 17 
    #postcode: 
    city: Berlin
    state: 10315 # Mix and match elements to get proper address
    country: Germany
footnote:
  - code: 1
    text:
  - code: 2
    text: "Current email address: [fay.webster@hu-berlin.de)](mailto:fay.webster@hu-berlin.de))"
abstract:
  - "Parasites in hybrid zones can give insight into species barriers, as they are modulating the fitness of hybrid hosts. Recent findings have demonstrated lower infection intensities with parasites in hybrids in the European House Mouse Hybrid zone (HMHZ), indicating higher disease resistance. However, tolerance has not yet been addressed in depth, as it is impractical to measure in wild populations. In an attempt to predict and evaluate the health impact of parasite infections and extrapolate tolerance in the HMHZ, we use a machine learning method. A random forest model was trained on immune parameters measured in experimental lab infections with Eimeria and then applied to data obtained from field sampling. Our predictions revealed that these infections are more detrimental to hybrid male mice. This approach represents an initial step in assessing tolerance in field studies."
keywords:
  - hybrids
  - parasites
#boxedlist: # Not fully compatible with all document formats
competing_interests: |
  There are no competing interest.
author_contributions: |
  To be worked on
acknowledgements: |
  To be worked on
bibliography: mybibfile.bib # Includes refs in OUP example template
#csl: https://www.zotero.org/styles/oxford-university-press-note
output:
  word_document: default
  pdf_document: default
  html_document: default
  rticles::oup_article:
    oup_version: 1 # 1 = 2020 CTAN OUP CLS package; 0 = 2009 OUP CLS package
    journal: 
    document_style: "contemporary" # Can be contemporary, modern, traditional
    papersize: "large" # Can be large, medium, small
    #citation_package: "default" # Uncomment when using a CSL; default "natbib"
    namedate: TRUE # Set FALSE to use numeric refs; Default FALSE
    #number_sections: FALSE # Uncomment to not number sections; default TRUE
    #number_lines: TRUE # Use `lineno` package - Default FALSE
    #number_lines_options: ["mathlines","switch"]  # Options for latex lineno package.
    #onecolumn: TRUE # Uncomment for one column format; default FALSE
    extra_dependencies:
      - booktabs # to use with knitr::kable() example below
#urlcolor: orange
#linkcolor: green
#citecolor: red
header-includes:
  #- \usepackage[nomarkers,tablesfirst]{endfloat} # For figures and tables at end
  - \theoremstyle{thmstyleone} # Theorem stuff from OUP template
  - \newtheorem{theorem}{Theorem} #  meant for continuous numbers. %%\newtheorem{theorem}{Theorem}[section] # meant for sectionwise numbers. optional argument [theorem] produces theorem numbering sequence instead of independent numbers for Proposition
  - \newtheorem{proposition}[theorem]{Proposition} # %%\newtheorem{proposition}{Proposition}" # to get separate numbers for theorem and proposition etc.
  - \theoremstyle{thmstyletwo}
  - \newtheorem{example}{Example}
  - \newtheorem{remark}{Remark}
  - \theoremstyle{thmstylethree}
  - \newtheorem{definition}{Definition}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = FALSE, # By default, hide code; set to TRUE to see code
  fig.pos = 'th', # Places figures at top or here
  out.width = '100%', dpi = 300, # Figure resolution and size
  fig.env="figure"
) # Latex figure environment

options(knitr.table.format = "latex") # For kable tables to write LaTeX table directly
```

```{r load libraries, message = FALSE, include=FALSE, warning=FALSE, echo=FALSE}
library(mice)
library(tidyr)
library(tidyverse)
library(VIM)
library(fitdistrplus)
library(fitur)
library(visdat)
library(corrplot)
library(dplyr)
```



# Introduction
## Introduce the topic

# Methods

## Laboratory infection experiments


###  Mouse strains (Luke)
In order to gain a better understanding of tolerance in hybrid mice we established a laboratory model of experimental lab infections with the Eimeria spp. Our experimental setup is a variation of the framework in Balard et al., 2020. The mice used are four wild-derived inbred mouse strains and their generated F1 hybrids. The mouse strains are fully inbred, as they have pased through at least 20 generations of sibling pairing. From the fours strains, two were used as a representation of the M. m. domesticus: SCHUNT (Locality: Schweben, Hessen, Germany [N: 5°0 26', E: 9°36'] @martincova2019phenotypic) and STRA (Locality: Straas, Bavaria, Germany [N: 50°11', E: 11°46'] (@pialek2008development). The two following strains were in turn derived from M. m. musculus: BUSNA (Locality: Buškovice, Bohemia, Czech Republic [N: 5°0 14', E: 1°3 22'] (pialek2008development)) and PWD (Locality: Kunratice, Bohemia, Czech Republic [N: 5°0 01', E: 14 2°9'] (@gregorova2000pwd). In our setup there are two two intersubspecific hybrids (STRAxBUSNA and SCHUNTxPWD) and two intrasubspecific hybrids (SCHUNTxSTRA and PWDxBUSNA). The mice were between 5.6 and 21.4 weeks. The mice were acquired from the Institute of Vertebrate Biology of the Czech Academy of Sciences in Studenec (license number 61974/2017-MZE-17214; for further details on strains see https://housemice.cz/en).

Infections with the parasite Eimeria induce a protective immune reaction in the host against reinfection (Rose et al., 1992a; Smith & Hayday, 2000). The feces of the naive mice were tested to ensure that the mice were Eimeria spp., prior to infection, following the methods of Balard @balard2020coupling. 

### Infections with Eimeria spp. (Luke)
The procedure used is as described in in Balard et al., 2020. During the infections mice were housed solo in cages. We infected the mice orally with 150 sporulated oocysts of one Eimeria isolate suspended in 100 µl phosphate-buffered saline (PBS). The mice had access to food and water ad libitum SNIFF, Rat/Mouse maintenance feed 10 mm and were observed daily for 8 days until their sacrifice by cervical dislocation. In the case that individual mice showed severe adverse health effects or extreme weigh loss of more than 18% relative to their weight at the start of experiment, were then sacrificed earlier at defined humane end points (experiment license Reg. 0431/17). Daily measurements of weight were recorded and fecal matter was gathered. Collected feces were supspended in 2% potassium dichromate and paaracite oocysts were retrieved by NaCl flotation. 

To enable a consistent distribution between experimental groups, mice were allocated at random, while ensuring a similar distribution of age and sex between groups. 

### Gene expression high-thoughput qPCR (Luke)
Homogenized caecum tissue was processed for RNA using the filter based innuPREP RNA Mini Kit 2.0 (Jena analytik, Germany) according to manufacturer instructions. Extracted RNA was quantified using NanoDrop 2000c (Thermo Scientific, Waltham, USA) and transcribed into cDNA with iScript (Bio-rad Laboratories, Hercules, California, United States), following manufacturer protocol.
Gene expression was ascertained via hight-throughput qPCR of cDNA from the caecum extracted material, plated onto Fluidigm IFC (integrated fluidic circuit), initialized by the Juno controller and read in the Fluidigm BioMark HD (PN 100-7222 C1, Fluidigm, South San Francisco, California, United States). After cDNA conversion mentioned above, the samples went through the recommended preparation steps, carried out in sterile extractor hood, with reagents being kept at 4 °C when in use, and at -20 °C when stored overnight. All pipetting was done using sterile filter tips. 
Table of wet-lab tested primers can be found in the appendix file 20614FDGP21T1_DESIGN

### Specific target amplification (STA) using Fluidigm PreAmp master mix (Luke)
ll primers at 100µM were pooled together in a micro centrifuge tube and resulting solution was diluted to a final concentration of 500 nM, with a 10 mM Tris-HCl (pH 8,0), 0.1 mM EDTA TE Buffer (12090-015, Invitrogen, Waltham, Massachusetts, United States). Fluidigm Preamp Master Mix (Fluidigm PN 100-5580) was added to the solution according to the manufacturer instructions, to create delta gene assay Preamp Master Mix. Sample cDNA (including a non-template control (NTC)) was plated onto 96-well plates and the master mix added to each well. Sample plate was then gently vortexed for 5 seconds and spun down at 1,000 x g for 1 minute. The amplification reactions were carried out in the Biometra TOne 96 (846-2-070-301, Analytic Jena, Jena, Germany) at the following cycling conditions: Hold at 95 ºC for 2 minutes, 95 ºC for 15 seconds, then 60 ºC for 4 minutes (15x), Hold at 4 ºC for infinity.

### Primers and sample preparation: (Luke)
1.5 µL of primer (wet lab tested) at 100 µM ,13.5 µL of TE Buffer and 15 µL of 2X Assay Loading Reagent (PN 100-7611, Fluidigm), were combined to create 10X Assay solutions. Sample reaction mixes were then created by combining 1.8 µL of preamplified and Exonuclease I treated cDNA, 2 µL of 2X SsoFast EvaGreen Supermix with low ROX™ (Bio-Rad, PN 172-5211) and 192.24 Delta Gene Sample Reagent (PN 100-6653, Fluidigm), to create 4 µL stock per 1 sample reaction. This was scaled to accommodate sample repeat runs. Prepared samples and primers were stored at -20ºC until the IFC chips were primed.
 
### IFC qPCR runs (Luke)
The 192.24 IFC was plated and treated as per manufacturer instructions (PN 100-7222 C1), consisting of control line fluid injection into accumulator 2 slot, removal of protective film, pipetting of samples and 10X assay mixes in 3 µL volumes, adding 150 µL of Actuation Fluid (PN 100-6250) into the P1 port, 150 µL of Pressure Fluid (PN 100-6249) into the P2 and P3 ports, 20 µL of Pressure Fluid into the P4 and P5 ports and finally initializing the IFC in the Juno controller, using the Load Mix 192.24 GE script. After the initialization completed, the IFC was transferred into the BioMark HD and the assay properties were set as follows:  Application type: Gene Expression, Passive reference: ROX, Assay: Single probe, Probe type: EvaGreen, using the GE 192x24 PCR+Melt v2.pcl protocol, on Auto-exposure. 

## Wild mice 

During the years 2016 to 2019, we sampled 1889 mice in the House Mouse Hybrid zone. We used live traps to capture mice in farms and houses, during September every year. To catch mice of *M. musculus musculus*, *M. musculus domesticus* and hybrid origin, we selected a large geographic area in Brandenburg and Neuruppin.  For each mouse, we have collected information during dissections, on their body length, their weight and their pathophysiology. Tissue samples were used to genotype the hosts. Further information on the set-up of the experiment can be found in previous publications of the group @balard2020intensity From 1889 mice, 336 were selected with a varying genotype for Fluidigm BioMark assay. Another 95 were selected for fluorescence-activated cell sorting. 

## Statistical Analysis

### Imputation of missing data 

```{r MICE_imputation, warning=FALSE, message=FALSE, include=FALSE}

setwd("~/GitHub/Eimeria_mouse_immunity/")

hm <- read.csv("output_data/1.MICE_cleaned_data.csv")

# Vectors for selecting genes
#Lab genes
# The measurements of IL.12 and IRG6 are done with an other assay and will 
#ignore for now
Gene_lab   <- c("IFNy", "CXCR3", "IL.6", "IL.13", "IL.10",
                "IL1RN","CASP1", "CXCL9", "IDO1", "IRGM1", "MPO", 
                "MUC2", "MUC5AC", "MYD88", "NCR1", "PRF1", "RETNLB", "SOCS1", 
                "TICAM1", "TNF") #"IL.12", "IRG6")
Genes_wild   <- c("IFNy", "CXCR3", "IL.6", "IL.13", "IL.10", 
                  "IL1RN","CASP1", "CXCL9", "IDO1", "IRGM1", "MPO", 
                  "MUC2", "MUC5AC", "MYD88", "NCR1", "PRF1", "RETNLB", "SOCS1", 
                  "TICAM1", "TNF") #, "IL.12", "IRG6")
Facs_lab <- c("CD4", "Treg", "Div_Treg", "Treg17", "Th1", 
                    "Div_Th1", "Th17", "Div_Th17", "CD8", "Act_CD8", 
                    "Div_Act_CD8", "IFNy_CD4", "IFNy_CD8","Treg_prop", 
                    "IL17A_CD4")  
Facs_wild <- c( "Treg", "CD4", "Treg17", "Th1", "Th17", "CD8")

# data imputation

# Genes
field <- hm %>%
  dplyr::filter(origin == "Field") 

field <- unique(field)

genes_mouse_field <- field %>%
  dplyr::select(c(Mouse_ID, all_of(Genes_wild)))

genes <- genes_mouse_field  %>%
  dplyr::select(-Mouse_ID)

#remove rows with only nas
genes <- genes[,colSums(is.na(genes))<nrow(genes)]

#remove colums with only nas 
genes <- genes[rowSums(is.na(genes)) != ncol(genes), ]

genes_mouse_field <- genes_mouse_field[row.names(genes), ]

##select same rows in the first table
field <- field[row.names(genes), ]

###############lab
#select the genes and lab muce
lab <- hm %>%
  dplyr::filter(origin == "Lab", Position == "mLN") #selecting for mln to avoid
# duplicates

lab <- unique(lab)

gene_lab_mouse <- lab %>%
  dplyr::select(c(Mouse_ID, all_of(Gene_lab))) 

gene_lab_mouse <- unique(gene_lab_mouse)

genes_lab <- gene_lab_mouse[, -1]

#remove rows with only nas
genes_lab <- genes_lab[,colSums(is.na(genes_lab))<nrow(genes_lab)]

#remove colums with only nas 
genes_lab <- genes_lab[rowSums(is.na(genes_lab)) != ncol(genes_lab), ]

genes_lab <- unique(genes_lab)

#select same rows in the first table
gene_lab_mouse <- gene_lab_mouse[row.names(genes_lab), ]

##select same rows in the first table
lab <- lab[row.names(genes_lab), ]

hm_genes <- rbind(gene_lab_mouse, genes_mouse_field)

hm_selection_g <- rbind(lab, field)

genes <- hm_genes %>%
 dplyr::select(-Mouse_ID)

# looking at patterns of nas
#pattern_na <-as.data.frame(md.pattern(field_genes))
sapply(hm_genes, function(x) sum(is.na(x)))
#had to remove as they were disturbing the imputation: Worms_presence, MC.Eimeria.FEC,  Heligmosomoides_polygurus, Zfy2, Y,  MpiC,
#vis_miss(field)
# The frequency distribution of the missing cases per variable can be obtained 
# as:
init <- mice(genes, maxit = 0)
#we want to impute only the specific variables
meth <- init$method

aggr_plot <- aggr(hm_genes, col=c('navyblue','red'), numbers=TRUE, sortVars=TRUE, labels=names(hm_genes), cex.axis=.7, gap=3, ylab=c("Histogram of missing data","Pattern"))

marginplot(hm_genes[c(4,5)])

# m=5 refers to the number of imputed datasets. Five is the default value.
igf <- mice(genes, m = 5, seed = 500) # method = meth,
summary(igf)
# to check each column with imputed data
## igf$imp$IFNy
#Now we can get back the completed dataset using the complete()
complete_genes <- complete(igf, 1)
#sapply(complete_field, function(x) sum(is.na(x)))
#visualize missingness
vis_dat(complete_genes)
#remove the non imputed genes from our data set
hm_selection_g <- hm_selection_g %>%
  dplyr::select(-c("IFNy", "CXCR3", "IL.6", "IL.13", "IL.10",
                "IL1RN","CASP1", "CXCL9", "IDO1", "IRGM1", "MPO", 
                "MUC2", "MUC5AC", "MYD88", "NCR1", "PRF1", "RETNLB", "SOCS1", 
                "TICAM1", "TNF"))
# add the new imputed genes to the data
hm_selection_g <- cbind(hm_selection_g, complete_genes)

xyplot(igf, IFNy ~ IL.13 + IRGM1 + MUC2, pch=18,cex=1)

xyplot(igf,IFNy ~ IL.10 + PRF1 + CASP1, pch=18,cex=1)

###############lab
#select the facs and lab muce
lab <- hm %>%
  dplyr::filter(origin == "Lab", Position == "mLN") #selecting for mln to avoid
# duplicates
lab <- unique(lab)
facs_mouse <- lab %>%
  dplyr::select(c(Mouse_ID, all_of(Facs_lab))) #choosing the same with the wild
facs_mouse <- unique(facs_mouse)
facs_lab <- facs_mouse[, -1]
#remove rows with only nas
facs_lab <- facs_lab[,colSums(is.na(facs_lab))<nrow(facs_lab)]
#remove colums with only nas 
facs_lab <- facs_lab[rowSums(is.na(facs_lab)) != ncol(facs_lab), ]
#select same rows in the first table
facs_mouse <- facs_mouse[row.names(facs_lab), ]
##select same rows in the first table
lab <- lab[row.names(facs_mouse), ]
#########################Field
###########field
# somehow the field samples have the origin na,
# fix that
field <- hm %>%
  dplyr::filter(origin == "Field") 

field <- unique(field)

facs_mouse <- field %>%
  dplyr::select(c(Mouse_ID, all_of(Facs_wild))) 

facs_field <- facs_mouse[,-1]

#remove rows with only nas
facs_field <- facs_field[,colSums(is.na(facs_field))<nrow(facs_field)]

#remove colums with only nas 
facs_field <- facs_field[rowSums(is.na(facs_field)) != ncol(facs_field), ]

##select same rows in the first table
field <- field[row.names(facs_field), ]

facs_data <- full_join(lab, field, by = intersect(colnames(lab), colnames(field)))

facs_data <- unique(facs_data) %>%
  dplyr::select(-c("IFNy", "CXCR3", "IL.6", "IL.13", "IL.10",
                "IL1RN","CASP1", "CXCL9", "IDO1", "IRGM1", "MPO", 
                "MUC2", "MUC5AC", "MYD88", "NCR1", "PRF1", "RETNLB", "SOCS1", 
                "TICAM1", "TNF"))

hm_selection_g <- hm_selection_g %>% 
  full_join(facs_data, by = intersect(colnames(facs_data), colnames(hm_selection_g)))

hm_selection_g <- unique(hm_selection_g)
```

To make the most of our data collection, we aimed to resolve missingness. Missing data were imputed using multiple imputations by chained equations. We used the package MICE in R @van2011mice, with five imputed data sets and five iterations. Data generated by FACS or the Gene Expression / Biomarker assay were regarded as missing if each mouse had measurements for some variables. For each continuous variable, we specified a predictive mean matching model. All the remaining variables were used as predictors in the imputation. To control the quality of our imputations, we evaluated the distribution plot of the existing data and the imputed data for all measurements \ref{fig:fig1} \ref{fig:fig2}. Further, we tested for convergence. We assume data is "missing completely at random" or "missing at random". For both types of missingness, multiple imputation is a suggested method to impute missing variables @van2018flexible. 

#### Questions / To-dos:
 
1. Should I log-transform the data prior to imputation?
2. Increasing produced data sets / iterations
3. sensitivity analyses using complete cases only 

### Random forest 

In an attempt to predict the health impact of infections in mice, we used a random forest model @breiman2001random. We chose the maximum weight loss during the experimental laboratory infections of mice with the parasite *Eimeria spp.* as a response variable. Maximum weight loss is here used as a proxy describing the health impact caused by infections. The random forest was constructed utilizing the expression data of 20 genes, obtained by the biomarker assay (utilizing the R package “randomForest,” ntree = 1000). The data set was split into a training data set of 70 % and to a testing data set of 30 % to avoid over-fitting and to assess the performance of the model on "unseen" data. As a quality assessment, we used k- fold cross validation, where the data set is divided into k subsets. Each time, the model is assessed by using one of the k subsets as the test data set and the other as the training data set. The model was then fitted on each k-subset and afterwards evaluated on the test set. Last the evaluation score was noted and the model then discarded.









```{r fig1, fig.cap="Stripplot of observed and imputed data", fig.env="figure*"}
stripplot(igf, pch = 20, cex = 1.2)
```

```{r fig2, fig.cap="Density plot of observed and imputed data",fig.env="figure*"}
densityplot(igf, height = 1000, width = 800)
```

```{r fig3, fig.cap="Inspecting to evaluate trace lines for convergence", fig.env="figure*", echo=FALSE, message = FALSE, include = FALSE}
plot(igf)
```







# Results 
# Discussion
# Conclusion

## A subsection



# Literature citations



# Equations

An equation without a label for cross-referencing:

$$
E=mc^2
$$

An inline equation: $y=ax+b$

An equation with a label for cross-referencing:

\begin{equation}\label{eq:eq1}
\int^{r_2}_0 F(r,\varphi){\rm d}r\,{\rm d}\varphi = 1
\end{equation}

This equation can be referenced as follows: Eq. \ref{eq:eq1}

# Inserting R figures

The code below creates a figure. The code is included in the output because `echo=TRUE`.

```{r figa, fig.cap="This is the first figure.",echo=TRUE}
plot(1:10,main="Some data",xlab="Distance (cm)",
     ylab="Time (hours)")
```

You can reference this figure as follows: Fig. \ref{fig:fig1}.

## Figures spanning two-columns

Figures can span two columns be setting `fig.env="figure*"`.

```{r figb, fig.cap="This is a wide figure.",fig.env="figure*"}
plot(1:5,pch=19,main="Some wide data",
     xlab="Distance (cm)",ylab="Time (hours)")
```

Reference to second figure: Fig. \ref{fig:fig2}

# Tables

## Generate a table using `xtable`

```{r xtabletable, results="asis", echo=TRUE}
df = data.frame(ID=1:3,code=letters[1:3])

# Creates tables that follow OUP guidelines 
# using xtable
library(xtable) 
print(xtable(df,caption="This is a xtable table.",
             label="tab:tab1"),
      comment=FALSE,caption.placement="top")
```

You can reference this table as follows: Table \ref{tab:tab1}.

## Generate a table using `kable`

```{r kabletable, echo=TRUE}
df = data.frame(ID=1:3,code=letters[1:3])

# kable can alse be used for creating tables
knitr::kable(df,caption="This is a kable table.",
             booktabs=TRUE,label="tab2")
```

You can reference this table as follows: Table \ref{tab:tab2}.

## Table spanning two columns

Tables can span two columns be setting `table.envir = "table*"` in `knitr::kable`.

```{r widetable, echo=TRUE}
df = data.frame(ID=1:3,code1=letters[1:3],
                code2=letters[4:6],
                code3=letters[7:9],
                code4=letters[10:12],
                code5=letters[13:15])

# kable can alse be used for creating tables
knitr::kable(df,caption="This is a wide kable table.",
             #format="latex",
             table.envir="table*",
             booktabs=TRUE,label="tab3")
```

# Cross-referencing sections

You can cross-reference sections and subsections as follows: Section \ref{literature-citations} and Section \ref{a-subsection}.

***Note:*** the last section in the document will be used as the section title for the bibliography.

For more portable and flexible referencing of sections, equations, figures and tables, use [`bookdown::pdf_document2`](https://github.com/rstudio/bookdown) with YAML header option `base_format: rticles::oup_article`.

# Appendices {-}

::: {.appendices latex=true}

# Section title of first appendix

blabla

## Subsection title of first appendix

and so on....
:::

# References



